code/streaming/Hacl.Streaming.Blake2.fst

```diff
@@ -733,49 +680,45 @@ let blake2 (a : alg) (m : valid_m_spec a)
            (init : blake2_init_st a m)
            (update_multi : blake2_update_multi_st a m)
            (update_last : blake2_update_last_st a m)
-           (finish : blake2_finish_st a m)
-           (no_key : bool) (key_size : key_size_t a no_key) :
+           (finish : blake2_finish_st a m) :
   I.block unit =
   I.Block
     I.Erased (* key management *)
     
-    (stateful_blake2 a m) (* state *)
-    (k a no_key key_size) (* key *)
+    (stateful_blake2 a m)     (* state *)
+    (I.stateful_unused unit)  (* key *)
```

Comment:
why do we still have an argument here? I thought the `key` was dropped everywhere...?

Reply:
ah ok I get it -- blake2 has lost the ability to operate in streaming + key mode, but poly1305 still can operate with a key in incremental mode

Reply:
Yes exactly : I removed the keyed mode for streaming blake2 and incremental blake2 *only*. Which means we still have an argument for the stateful key in the streaming functor (for poly1305), which in the very specific case of blake2 we instantiate with `stateful_unused`.